### PR TITLE
feat: sync device contacts to cozy

### PR DESCRIPTION
**Pull request by y-lohse**
Demo: https://thumbs.gfycat.com/PepperyAltruisticBluebird-mobile.mp4

The code here is ready for a review, but before merging we'll need a new version of cozy client js that includes [this PR](https://github.com/cozy/cozy-client-js/pull/192)

### CODE REVIEW

**Comment by y-lohse** [comment]
[(5) [FIL] Importer les contacts du téléphone Android et iOS](https://trello.com/c/akcF4891/350-5-fil-importer-les-contacts-du-t%C3%A9l%C3%A9phone-android-et-ios)

File: mobile/src/actions/contactsBackup.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  let deviceContacts
+
+  try {
+    deviceContacts = await loadDeviceContacts()
+  } catch (e) {
+    // the authorization has been revoked
+    dispatch(setBackupContacts(false))
+    return
+  }
+  // keep only contacts with an email address
```
Comment (enguerran) [inline mobile/src/actions/contactsBackup.js]: why don't you create a named function to get such contacts?

```js
function filterContactsWithEmailAddress (contacts) {
   return contacts.filter()
}
```

Then, first you don't need a comment line to express the intention. Second you don't add out-of-scope responsibility to the main function `backupContacts`.

Comment (y-lohse) [inline mobile/src/actions/contactsBackup.js]: Yup, did that 👍

File: mobile/src/actions/contactsBackup.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        {
+          filter: (new Date()).toString(), // we just want a filter that won't match anything
+          multiple: false
+        }
+      )
+    }
+  })
+}
+
+// versions 0.3.5 of the app and prior only requested permissions for io.cozy.files ; this function makes checks if we have permissions for io.cozy.contact and requests them if not
```
Comment (y-lohse) [inline mobile/src/actions/contactsBackup.js]: I strongly disagree with that :-)

**Comment by y-lohse** [comment]
@kosssi thanks for the reminder! I added it manually in the `config.xml`since the `cordova add` command didn't do anything. Is that normal?

**Pull request merged**